projects/103_hmwrk/src/main.c

```diff
@@ -0,0 +1,55 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress adc_addr = {
+  .port = GPIO_PORT_A,
+  .pin = 6,
+};
+
+static GpioAddress button_addr = {
+  .port = GPIO_PORT_B,
+  .pin = 2,
+};
+
+static InterruptSettings s_interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+void read_from_adc(const GpioAddress *address, void *context) {
+  uint16_t data = 0;
+  if (adc_read_raw_pin(*(GpioAddress *)context, &data) == STATUS_CODE_OK) {
```

Comment:
This cast is a bit awkward. The format we generally use is at the top of the function:
```
GpioAddress *address = (GpioAddress*)context;
```
and then access all fields from the pointer, ie. `address->port`
This isn't a functional problem, more of a best coding practices.